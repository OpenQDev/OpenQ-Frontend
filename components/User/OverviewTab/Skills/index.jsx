import React from 'react';
import Roles from '../Roles';

const Skills = ({ user }) => {
  return (
    <div className='px-8 py-6 pb border-t border-web-gray'>
      <h2 className='font-semibold text-lg pb-2 '>Skills</h2>
      <ul>
        <Roles user={user} category='Dev Role' defaultRoles={user.devRoles} />
        <Roles user={user} category='Other Role' defaultRoles={user.otherRoles} />
        <Roles user={user} category='Language' defaultRoles={user.languages} />
        <Roles user={user} category='Framework' defaultRoles={user.frameworks} />
      </ul>
    </div>
  );
};

export default Skills;
