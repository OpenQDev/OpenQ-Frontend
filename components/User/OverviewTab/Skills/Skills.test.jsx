import React from 'react';
import Skills from '../../../../components/User/OverviewTab/Skills';
import ShallowRenderer from 'react-test-renderer/shallow';
import Constants from '../../../../test-utils/constant';

describe('Skills', () => {
  const user = {
    ...Constants.accountData,
    devRoles: ['Junior Developer'],
    otherRoles: [''],
    languages: ['JavaScript', 'Solidity'],
    frameworks: ['React'],
  };
  it('should match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<Skills user={user} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
