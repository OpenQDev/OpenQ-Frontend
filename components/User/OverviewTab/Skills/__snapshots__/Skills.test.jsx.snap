// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skills > should match DOM Snapshot 1`] = `
<div
  className="px-8 py-6 pb border-t border-web-gray"
>
  <h2
    className="font-semibold text-lg pb-2 "
  >
    Skills
  </h2>
  <ul>
    <Roles
      category="Dev Role"
      defaultRoles={
        [
          "Junior Developer",
        ]
      }
      user={
        {
          "devRoles": [
            "Junior Developer",
          ],
          "email": "abc123@gmail.com",
          "frameworks": [
            "React",
          ],
          "github": "MDQ6VXNlcjcyMTU2Njc5",
          "id": "63a3a266ab24d99d359a6d51",
          "languages": [
            "JavaScript",
            "Solidity",
          ],
          "otherRoles": [
            "",
          ],
          "starredOrganizationIds": undefined,
          "twitter": "https://twitter.com/Christo28120856",
          "username": "Christopher",
          "watchedBounties": undefined,
        }
      }
    />
    <Roles
      category="Other Role"
      defaultRoles={
        [
          "",
        ]
      }
      user={
        {
          "devRoles": [
            "Junior Developer",
          ],
          "email": "abc123@gmail.com",
          "frameworks": [
            "React",
          ],
          "github": "MDQ6VXNlcjcyMTU2Njc5",
          "id": "63a3a266ab24d99d359a6d51",
          "languages": [
            "JavaScript",
            "Solidity",
          ],
          "otherRoles": [
            "",
          ],
          "starredOrganizationIds": undefined,
          "twitter": "https://twitter.com/Christo28120856",
          "username": "Christopher",
          "watchedBounties": undefined,
        }
      }
    />
    <Roles
      category="Language"
      defaultRoles={
        [
          "JavaScript",
          "Solidity",
        ]
      }
      user={
        {
          "devRoles": [
            "Junior Developer",
          ],
          "email": "abc123@gmail.com",
          "frameworks": [
            "React",
          ],
          "github": "MDQ6VXNlcjcyMTU2Njc5",
          "id": "63a3a266ab24d99d359a6d51",
          "languages": [
            "JavaScript",
            "Solidity",
          ],
          "otherRoles": [
            "",
          ],
          "starredOrganizationIds": undefined,
          "twitter": "https://twitter.com/Christo28120856",
          "username": "Christopher",
          "watchedBounties": undefined,
        }
      }
    />
    <Roles
      category="Framework"
      defaultRoles={
        [
          "React",
        ]
      }
      user={
        {
          "devRoles": [
            "Junior Developer",
          ],
          "email": "abc123@gmail.com",
          "frameworks": [
            "React",
          ],
          "github": "MDQ6VXNlcjcyMTU2Njc5",
          "id": "63a3a266ab24d99d359a6d51",
          "languages": [
            "JavaScript",
            "Solidity",
          ],
          "otherRoles": [
            "",
          ],
          "starredOrganizationIds": undefined,
          "twitter": "https://twitter.com/Christo28120856",
          "username": "Christopher",
          "watchedBounties": undefined,
        }
      }
    />
  </ul>
</div>
`;
